api/app/signals/apps/feedback/serializers.py

```diff
@@ -26,6 +28,38 @@ class Meta:
         }
 
     def update(self, instance, validated_data):
+        # TODO: consider whether using a StandardAnswer while overriding the
+        # is_satisfied field should be considered an error condition and return
+        # an HTTP 400.
         validated_data['submitted_at'] = timezone.now()
 
+        # Check whether the relevant Signal instance should possibly be
+        # reopened (i.e. transition to VERZOEK_TOT_HEROPENEN state).
+        is_satisfied = validated_data['is_satisfied']
+
+        if is_satisfied:
+            reopen = False
+        else:
+            feedback_text = validated_data['text']
+
+            try:
+                sa = StandardAnswer.objects.get(text=feedback_text)
+            except StandardAnswer.DoesNotExist:
+                reopen = True
+            else:
+                if not sa.is_satisfied:
+                    reopen = sa.reopens_when_unhappy
+
+        # Reopen the Signal (melding) if need be.
+        if reopen:
+            signal = Signal.objects.get(pk=instance._signal_id)
+
+            # Only allow a request to reopen when in state workflow.AFGEHANDELD
+            if signal.status.state == workflow.AFGEHANDELD:
+                payload = {
+                    'text': 'De melder is niet tevreden blijkt uit feedback. Zo nodig heropenen.',
+                    'state': workflow.VERZOEK_TOT_HEROPENEN,
+                }
+                Signal.actions.update_status(payload, signal)
```

Comment:
This could be an action on the signal actions manager? Something like `Signal.actions.request_to_reopen(signal)`

Reply:
Maybe, but I would like to hold off until we know this stuff is actually what the business wants. We also have to be careful what we add to the actions manager --- it very central to SIA.